codeMetadata.go

```diff
@@ -0,0 +1,37 @@
+package vmcommon
+
+const lengthOfCodeMetadata = 2
+
+const (
+	METADATA_UPGRADEABLE = 1
+	METADATA_RESERVED_1  = 2
+	METADATA_RESERVED_2  = 4
+	METADATA_RESERVED_3  = 8
+)
+
+// CodeMetadata represents smart contract code metadata
+type CodeMetadata struct {
```

Comment:
Private class with public methods to get?

Reply:
It can stay fully public. The info in its fields is static and straightforward - no getters needed.

Reply:
Yes, it stays public. It is used as arguments to other functions. It is returned by `parser.GetCodeMetadata()`. Won't make private. It is the simplest form of abstraction.